Content.Shared/_DV/CosmicCult/Components/MonumentComponent.cs

```diff
@@ -0,0 +1,149 @@
+    public int CurrentProgress;
+
+    /// <summary>
+    /// how much progress the cult need to make to tier up
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public int TargetProgress;
+
+    /// <summary>
+    /// offset used to make the progress bar reset to 0 every time
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public int ProgressOffset;
+
+    /// <summary>
+    /// A bool we use to set whether The Monument's UI is available or not.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public bool Enabled = true;
+
+    /// <summary>
+    /// how long the monument takes to transform on a tier up
+    /// </summary>
+    [DataField]
+    public TimeSpan TransformTime = TimeSpan.FromSeconds(2.8);
+
+    /// <summary>
+    /// the entity for the currently scribed glyph
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public EntityUid? CurrentGlyph;
+
+    /// <summary>
+    /// the timer used for ticking healing from vacuous vitality
+    /// </summary>
+    [AutoPausedField, DataField(customTypeSerializer: typeof(TimeOffsetSerializer))]
+    public TimeSpan CheckTimer = default!;
+
+    /// <summary>
+    /// the amount of time between the above timer's ticks
+    /// </summary>
+    [DataField]
+    public TimeSpan CheckWait = TimeSpan.FromSeconds(5);
+
+    /// <summary>
+    /// Passive healing factor for cultists w/ the ability near the monument
+    /// </summary>
+    [DataField]
+    public DamageSpecifier MonumentHealing = new()
+    {
+        DamageDict = new()
+        {
+            { "Blunt", 2},
+            { "Slash", 2 },
+            { "Piercing", 2 },
+            { "Heat", 2},
+            { "Shock", 2},
+            { "Cold", 2},
+            { "Poison", 2},
+            { "Radiation", 2},
+            { "Asphyxiation", 2 }
+        }
+    };
+
+    /// <summary>
+    /// wether or not there's a stage change queued
+    /// </summary>
+    [DataField]
+    public bool CanTierUp = true;
+
+    [DataField(customTypeSerializer: typeof(TimeOffsetSerializer)), AutoPausedField]
+    public TimeSpan? PhaseOutTimer;
+}
+
+[Serializable, NetSerializable]
+public sealed class InfluenceSelectedMessage(ProtoId<InfluencePrototype> influenceProtoId, NetEntity? sender) : BoundUserInterfaceMessage
+{
+    public ProtoId<InfluencePrototype> InfluenceProtoId = influenceProtoId;
+    public NetEntity? Sender = sender;
+}
+
+[Serializable, NetSerializable]
+public sealed class GlyphSelectedMessage(ProtoId<GlyphPrototype> glyphProtoId) : BoundUserInterfaceMessage
+{
+    public ProtoId<GlyphPrototype> GlyphProtoId = glyphProtoId;
+}
+
+[Serializable, NetSerializable]
+public sealed class GlyphRemovedMessage : BoundUserInterfaceMessage;
+
+[Serializable, NetSerializable]
+public enum MonumentVisuals : byte
+{
+    Monument,
+    Transforming,
+    FinaleReached,
+    FinaleActive,
+    Tier3,
+}
+
+[Serializable, NetSerializable]
+public enum MonumentVisualLayers : byte
+{
```

Comment:
FinaleLayer is referenced in the prototype yaml but never in code, and I don't know how visuals work so I don't know if that's correct or not

Reply:
@AftrLite what are you intending those sprites to be used for

Reply:
errr these are used, they're the sprites for the monument charging up for the finale
the value propagates through FinaleReached and is used to adjust that layer